src/sentry/sentry_metrics/indexer/mock.py

```diff
@@ -1,41 +1,44 @@
 from typing import Optional
 
-from sentry.models import Organization
+from django.conf import settings
+
+from sentry.utils.redis import redis_clusters
 
 from .base import StringIndexer, UseCase
 
-_STRINGS = {
-    "abnormal": 0,
-    "crashed": 1,
-    "environment": 2,
-    "errored": 3,
-    "healthy": 4,
-    "production": 5,
-    "release": 6,
-    "session.duration": 7,
-    "session.status": 8,
-    "session": 9,
-    "staging": 10,
-    "user": 11,
-}
-_REVERSE = {v: k for k, v in _STRINGS.items()}
+
+def get_client():
+    return redis_clusters.get(settings.SENTRY_METRICS_INDEXER_REDIS_CLUSTER)
 
 
 class MockIndexer(StringIndexer):
     """
     Mock string indexer
     """
 
-    def record(self, organization: Organization, use_case: UseCase, string: str) -> int:
-        """Mock indexer cannot record."""
-        raise NotImplementedError()
+    def record(self, org_id: str, use_case: UseCase, string: str) -> int:
+        """
+        If key already exists, grab that value, otherwise record both the
+        string to int and int to string relationships.
+        """
+        client = get_client()
 
-    def resolve(self, organization: Organization, use_case: UseCase, string: str) -> Optional[int]:
```

Comment:
Contrary to what I said previously, I think it would make sense to keep the `resolve` method. It's already in use [here](https://github.com/getsentry/sentry/blob/e59971700f9f1c1778030cc3944805d845a79e78/src/sentry/snuba/metrics.py#L422), and if the indexer entries ever get a TTL, it would not make sense to prolong the retention every time the indexer is queried from the product side.